src/NServiceBus.RabbitMQ/RabbitMQTransportInfrastructure.cs

```diff
@@ -25,13 +25,24 @@ public RabbitMQTransportInfrastructure(SettingsHolder settings, string connectio
             var connectionConfiguration = new ConnectionStringParser(settings).Parse(connectionString);
             connectionFactory = new ConnectionFactory(connectionConfiguration);
 
+            SetUsePublisherConfirmsIfPresent(settings, connectionConfiguration);
+
             CreateTopology();
 
             channelProvider = new ChannelProvider(connectionFactory, routingTopology, connectionConfiguration.UsePublisherConfirms);
 
             RequireOutboxConsent = false;
         }
 
+        static void SetUsePublisherConfirmsIfPresent(SettingsHolder settings, ConnectionConfiguration connectionConfiguration)
+        {
+            if (settings.HasSetting(SettingsKeys.UsePublisherConfirms))
+            {
+                var publisherCofirms = settings.Get<bool>(SettingsKeys.UsePublisherConfirms);
+                connectionConfiguration.UsePublisherConfirms = publisherCofirms;
+            }
```

Comment:
not having the full context, but there is some important and very weird thing about the default values on the settings holder:

* If you do a Get, it already returns values configured as default values when no other value has been set.
* If you do a GetOrDefault, it will return `default(T)` when no custom or "default" value has been found.

So you could already write:

`settings.GetOrDefault<bool>(SettingsKeys.UsePublisherConfirms)` which will return `false` if no value has been configured.

But on the other side, yes, the SettingsHolder class is intended and available to be used for storing settings so you don't necessarily need to run your own.